src/Game/GamePage.js

```diff
@@ -34,23 +35,52 @@ const GamePage = (props) => {
     fetchHitpoints();
   }, []);
 
-  useEffect(() => {
-    console.log("total player hp state: ", totalPlayersHp);
+  const myPlayerData = useMemo(() => {
+    return totalPlayersHp.find((playerData) => playerData.name === player);
   }, [totalPlayersHp]);
 
+  const isPlayerEliminated = myPlayerData && myPlayerData.Hitpoints === 0;
+
+  const checkWinCondition = () => {
+    let hasPlayerWon = false;
+    const eliminatedPlayers = totalPlayersHp.filter(
+      (playerData) => playerData.Hitpoints === 0
+    );
+    if (
+      eliminatedPlayers.length === totalPlayersHp.length - 1 &&
+      totalPlayersHp.length !== 0 &&
+      isPlayerEliminated === false
+    ) {
+      hasPlayerWon = true;
+    }
+    return hasPlayerWon;
+  };
+
+  const hasWon = useMemo(checkWinCondition, [totalPlayersHp]);
+
   const updateHitpointValues = (currentState, payload) => {
     const changeHitpoints = currentState.map((playerData) => {
       if (payload.new.name === playerData.name) {
-        return {
-          name: payload.new.name,
-          Hitpoints: payload.new.Hitpoints,
-        };
+        if (payload.new.Hitpoints > 0) {
+          return {
+            name: payload.new.name,
+            Hitpoints: payload.new.Hitpoints,
+          };
+        } else {
+          return {
+            name: payload.new.name,
+            Hitpoints: 0,
+          };
+        }
```

Comment:
```suggestion
        return {
          name: payload.new.name,
          Hitpoints: payload.new.Hitpoints > 0 ? payload.new.Hitpoints: 0,
        }
```
Not sure if hiding the actual hitpoints necessary but this is a cleaner way of writing what you're doing.